persistence/test/generic_test.go

```diff
@@ -34,6 +36,92 @@ func NewTestGenericActor[T any](protocolActorSchema schema.ProtocolActorSchema,
 	}
 }
 
+func getAllActorsTest[T any](
+	t *testing.T,
+	db *persistence.PostgresContext,
+	getAllActors func(height int64) ([]*T, error),
+	createTestActor func(*persistence.PostgresContext) (*T, error),
+	updateActor func(*persistence.PostgresContext, *T) error,
+	initialCount int,
+) {
+	// The default test state contains `initialCount` actors
+	actors, err := getAllActors(0)
+	require.NoError(t, err)
+	require.Len(t, actors, initialCount)
+
+	// Add 2 actors at height 1
+	db.Height++
+	_, err = createTestActor(db)
+	require.NoError(t, err)
+	_, err = createTestActor(db)
+	require.NoError(t, err)
+
+	// Check height 0
+	actors, err = getAllActors(0)
+	require.NoError(t, err)
+	require.Len(t, actors, initialCount)
+
+	// Check height 1
+	actors, err = getAllActors(1)
+	require.NoError(t, err)
+	require.Len(t, actors, initialCount+2)
+
+	// Add 1 actor at height 3
+	db.Height++
+	db.Height++
+	_, err = createTestActor(db)
+	require.NoError(t, err)
+
+	// Check height 0
+	actors, err = getAllActors(0)
+	require.NoError(t, err)
+	require.Len(t, actors, initialCount)
+
+	// Check height 1
+	actors, err = getAllActors(1)
+	require.NoError(t, err)
+	require.Len(t, actors, initialCount+2)
+
+	// Check height 2
+	actors, err = getAllActors(2)
+	require.NoError(t, err)
+	require.Len(t, actors, initialCount+2)
+
+	// Check height 3
+	actors, err = getAllActors(3)
+	require.NoError(t, err)
+	require.Len(t, actors, initialCount+3)
+
+	// Update the service nodes at different heights and confirm that count does not change
+	for _, actor := range actors {
+		db.Height++
+		err = updateActor(db, actor)
+		require.NoError(t, err)
+
+		// Check that count did not change
+		actors, err := getAllActors(db.Height)
+		require.NoError(t, err)
+		require.Len(t, actors, initialCount+3)
+	}
+
+	// Check height 1
+	actors, err = getAllActors(1)
+	require.NoError(t, err)
+	require.Len(t, actors, initialCount+2)
+
+	// Check height 10
+	actors, err = getAllActors(10)
+	require.NoError(t, err)
+	require.Len(t, actors, initialCount+3)
+
+	// DISCUSS_IN_THIS_COMMIT: Since we do not support `DeleteActor`, should we filter here based on status? If so, tests need to be updated.
```

Comment:
(Non-Actionable) 'Delete' is in limbo for now. I say let's document/leave it

Reply:
Comment updated.